homeassistant/helpers/aiohttp_client.py

```diff
@@ -0,0 +1,118 @@
+"""Helper for aiohttp webclient stuff."""
+import asyncio
+
+import aiohttp
+
+from homeassistant.core import callback
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+
+
+DATA_CONNECTOR = 'aiohttp_connector'
+DATA_CONNECTOR_NOTVERIFY = 'aiohttp_connector_notverify'
+DATA_CLIENTSESSION = 'aiohttp_clientsession'
+DATA_CLIENTSESSION_NOTVERIFY = 'aiohttp_clientsession_notverify'
+
+
+@callback
+def async_get_clientsession(hass, verify_ssl=True):
+    """Return default aiohttp ClientSession.
+
+    This method must be run in the event loop.
+    """
+    if verify_ssl:
+        key = DATA_CLIENTSESSION
+    else:
+        key = DATA_CLIENTSESSION_NOTVERIFY
+
+    if key not in hass.data:
+        connector = _async_get_connector(hass, verify_ssl)
+        clientsession = aiohttp.ClientSession(
+            loop=hass.loop,
+            connector=connector
+        )
+        _async_register_clientsession_shutdown(hass, clientsession)
+        hass.data[key] = clientsession
+
+    return hass.data[key]
+
+
+@callback
+def async_create_clientsession(hass, verify_ssl=True, auto_cleanup=True,
+                               **kwargs):
+    """Create a new ClientSession with kwargs, i.e. for cookies.
+
+    Is auto_cleanup False, you need call detach() after this session will be
+    unused. True it do it self on homeassistant_stop.
```

Comment:
'If auto_cleanup is False, you need to call detach() after the session returned is no longer used. Default is True, the session will be automatically detached on homeassistant_stop.'